include/nanocbor/nanocbor.h

```diff
@@ -201,6 +201,32 @@ int nanocbor_get_type(const nanocbor_value_t *value);
  */
 bool nanocbor_at_end(const nanocbor_value_t *it);
 
+/**
+ * @brief Retrieve a positive integer as uint8_t from the stream
+ *
+ * The resulting @p value is undefined if the result is an error condition
+ *
+ * @param[in]   cvalue  CBOR value to decode from
+ * @param[out]  value   returned positive integer
+ *
+ * @return              number of bytes read
+ * @return              negative on error
```

Comment:
Could you please extend the docs here to mention that the result is an error code if the decoded value is greater than 255 or spans more than 1 byte?

Same for the other decode functions of course :)

Reply:
I added a note in the comment.